File: Sorting Algorithms/Selection Sort/SelectionSort.c
/* C program for Selection Sort Algorithms*/

#include<stdio.h>
// swap using poiter
void swap(int *a, int *b)
{
    int temp = *b;
    *b = *a;
    *a = temp;
}

void SelectionSort(int array[],int size)
{
    int i, j, min_index;
    for(i = 0; i< size - 1; i++)
    {
        min_index = i;
        for(j = i+1 ; j < size ; j++)
        {
            if(array[min_index] > array[j])
                min_index = j;
        }

        /* 
        
        int temp = array[min_index];
         array[min_index] = array[i];
         array[i]= temp;
        
        */

        swap(&array[i], &array[min_index]);

    }
}


void printArray(int array[],int size)
{
    for(int i = 0; i < size; i++)
    {
        printf("%d  ",array[i] );
    }
}




int main()
{
    int data[] = {64, 12, 22, 25, 11, 23, 100, 99};
    int size = sizeof(data)/sizeof(data[0]);
    SelectionSort(data,size);
    printArray(data, size);
}

File: Sorting Algorithms/Selection Sort/SelectionSort.java
package javacode.sort;

import java.util.Arrays;

public class Selection_sort {
	public  void Select_sorted(int array[]) {
		int i,j, min_index;
		int n = array.length;
		
		for (i = 0; i < n-1; i++) {
			min_index = i;
			for (j = i+1; j < n; j++) {
				if(array[min_index]>array[j]) {
					min_index = j;	
				}
			}
			// after for loop for j swap array[i] & array[min_index]
			int temp = array[min_index];
			array[min_index]= array[i];
			array[i] = temp;
		}
			
	}
	
	public static void main(String[] args) {	
		int[] data = {64, 12, 22, 25, 11, 23, 100, 99};
		
		Selection_sort s_sorted =  new Selection_sort();
		s_sorted.Select_sorted(data);
		System.out.println(Arrays.toString(data));
		
	}

	
}
